samcli/commands/_utils/custom_options/hook_package_id_option.py

```diff
@@ -50,6 +50,16 @@ def handle_parse_result(self, ctx, opts, args):
                     raise click.BadParameter(
                         f"Parameters {opt_name}, and {','.join(self._invalid_coexist_options)} can not be used together"
                     )
+
+            # validate build-image is provided in case of build using container
+            # add this validation here to avoid running hook prepare and there is issue
+            if command_name == "build" and opts.get("use_container") and not opts.get("build_image"):
+                msg = (
+                    "Missing required parameter, need the --build-image parameter in order to use --use-container flag "
```

Comment:
Missing required parameter "--build-image", should that suffice?

Reply:
hmmm ... will it be clear for the customer .. I like explaining things :D

Reply:
It should be, since there is context on the user who just ran the command with a certain number of options. I like the explanation aspect, but if I wonder if we should start codifying it so that its consistent across commands.